common/src/format.rs

```diff
@@ -430,11 +432,17 @@ impl FormatSpec {
             | Some(FormatType::Octal)
             | Some(FormatType::Hex(_))
             | Some(FormatType::String)
-            | Some(FormatType::Character) => {
+            | Some(FormatType::Character)
+            | Some(FormatType::Number(Case::Upper)) => {
                 let ch = char::from(self.format_type.as_ref().unwrap());
                 Err(FormatSpecError::UnknownFormatCode(ch, "float"))
             }
-            Some(FormatType::Number) => Err(FormatSpecError::NotImplemented('n', "float")),
+            Some(FormatType::Number(Case::Lower)) => {
```

Comment:
By document, `n` seems to same as `g` except for seperators.
Can we reuse `g` implementation? 

https://docs.python.org/3/library/string.html#formatspec

Otherwise, if it is a sort of scientific notation, can we reuse `:e` format in Rust?

Reply:
Reused g's implementation and everything lines up for me locally but I'm not 100% sure how to test the locale based separators. Gonna see what the workflow thinks when it runs the tests.

Reply:
triggered the run now, locale based separators could be a future addition (i.e no need to do everything in one PR :smile:)

Reply:
> i.e no need to do everything in one PR 😄

Rome wasn't built in a day.